faux-mgs/src/main.rs

```diff
@@ -589,7 +590,17 @@ async fn main() -> Result<()> {
             .await?;
 
             // If usart::run() returns, the user detached; exit.
-            return Ok(());
+            //
+            // We don't just `return Ok(())` here because we'll bump into
```

Comment:
Nice workaround!

Reply:
Credit to @dancrossnyc! I brought this up during last week's Rust study group, and he both immediately recognized the problem (before we found the relevant tokio issue) and offered the suggestion to explicitly exit.